deepfence_ui/app/scripts/components/vulnerability-view/cve-image-report-container.js

```diff
@@ -14,46 +16,65 @@ class CVEImageReportContainer extends React.PureComponent {
     this.handleDownload = this.handleDownload.bind(this);
   }
 
-  handleDownload(params) {
-    const {
-      xlsxReportDownloadAction: action,
-    } = this.props;
+  async handleDownload(params) {
+    const { dispatch } = this.props;
 
-    const {
-      scanId = '',
-      nodeType = '',
-    } = params;
+    const { scanId = '', nodeType = '' } = params;
 
     const apiParams = {
       action: 'download_report',
       node_type: nodeType,
+      add_hist: false,
       action_args: {
         resources: [
           {
             type: 'cve',
             filter: {
-              scan_id: [
-                scanId,
-              ]
-            }
-          }
+              scan_id: [scanId],
+            },
+          },
         ],
         filters: {
-          type: [
-            nodeType,
-          ]
-        }
-      }
+          type: [nodeType],
+        },
+      },
     };
 
-    return action(apiParams);
+    const generateReport = dispatch(reportGenerateAction(apiParams));
+    dispatch(toaster('Generating Report ...'));
+    generateReport.then(res => {
+      if (res.success) {
+        const fetchReportStatus = dispatch(reportDownloadStatusAction());
+        fetchReportStatus.then(async res => {
+          if (res.data[0]) {
+            await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
why artificial delay?

Reply:
I added this because the report is taking at least a second to fully generate before we can download it. If the delay isn't there, the report is not getting downloaded because it is not fully generated yet.

Reply:
@saiprasanth1303 I think hardcoding 2 second wait won't work in all cases. Report generation time is dependent on the amount of data we process / no of pages. So it can take > 2 seconds if the image / host has 1000's of vulnerabilities.
Can you add 3 retries and then throw error?

Reply:
@ramanan-ravi Shall I add the 3 retries after the 2-second wait?

Reply:
Yes. Also 2 second wait between retries.